### PR TITLE
fix[examples]: update example cache-handler-redis

### LINKED ISSUE #66812: Redis cache handler example incorrect
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.2.0: Wed Nov 15 21:55:06 PST 2023; root:xnu-10002.61.3~2/RELEASE_ARM64_T6020
  Available memory (MB): 32768
  Available CPU cores: 12
Binaries:
  Node: 18.17.0
  npm: 9.6.7
  Yarn: 1.22.19
  pnpm: 8.11.0
Relevant Packages:
  next: 14.2.3 // There is a newer version (14.2.4) available, upgrade recommended! 
  eslint-config-next: 14.2.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.5
Next.js Config:
  output: standalone
 ⚠ There is a newer version (14.2.4) available, upgrade recommended! 
   Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
   Read more - https://nextjs.org/docs/messages/opening-an-issue
```

### Which example does this report relate to?

cache-handler-redis

### What browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

### Describe the Bug

The Cache handler return value is not compatible with the `@neshca/cache-handler` package.
Trying to run `next build` with the current logic throws a `TypeError: Cannot read properties of undefined (reading 'filter')`

### Expected Behavior

The [return value of the CacheHandler](https://github.com/vercel/next.js/blob/canary/examples/cache-handler-redis/cache-handler.js#L38) should be updated to the following (or similar):
```js
  /** @type {import("@neshca/cache-handler").CacheHandlerConfig} */
  return {
    handlers: [redisCache, localCache],
  };
```

### To Reproduce

* Create a cache-handler file with the current example code
* Run a next build
* The build will fail
* Change the cache handler return value from `cache -> handlers`
* The build will succeed

### PR DESCRIPTION
**Pull request by mauroaccornero**
### What?

Fix cache-handler-redis example with working cache handler 

### Why?

It was not working because next-shared-cache was updated but some code inside the cache-handler.js was deprecated after version 1.0.0.

When starting the app without Redis the cache handler was not switching to the LRU handler.

### How?

- Updated cache-handler-redis example with working cache handler from [next-shared-cache's example](https://caching-tools.github.io/next-shared-cache/redis)
- Add check for PHASE_PRODUCTION_BUILD to avoid [issues while building ](https://github.com/caching-tools/next-shared-cache/issues/284#issuecomment-1919145094)
- Add throw error for Redis error to switch to the LRU handler
- Updated dependencies
- Updated readme broken links
- Remove "version" from docker-compose's file because it's outdated [version outdated](https://github.com/docker/compose/issues/11628)

Closes NEXT-66812
Fixes #66812

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: bf8ce9b11d97d19c96b5912d4930aef691b435e8

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: bf8ce9b11d97d19c96b5912d4930aef691b435e8

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: examples/cache-handler-redis/cache-handler.js [hunk 1]
```diff
@@ -1,44 +1,77 @@
...
 
-  const localCache = createLruCache({
-    useTtl,
-  });
-
   return {
-    cache: [redisCache, localCache],
-    // read more about useFileSystem limitations https://caching-tools.github.io/next-shared-cache/configuration/use-file-system
-    useFileSystem: false,
+    handlers: [handler],
```
Comment (papertiger12) [inline examples/cache-handler-redis/cache-handler.js:74]: i think its helpful to show setting 2 handlers
```
handlers: [redisCache, localCache],
```

Comment (mauroaccornero) [reply examples/cache-handler-redis/cache-handler.js]: @papertiger12  Sure. I've just updated the handlers's array to use both handlers.  Thanks for the feedback

**Review by papertiger12 — APPROVED** [review]

**Review by delbaoliveira — APPROVED** [review]

**Comment by delbaoliveira** [comment]
Thank you @mauroaccornero 🙏🏼

**Pull request merged**